lib/sycamore/sycamore/connectors/pinecone/pinecone_scan.py

```diff
@@ -21,10 +21,14 @@ def execute(self) -> Dataset:
             for pids in index.list(namespace=self._namespace):
                 ids.extend(pids)
             for id, data in dict(index.fetch(ids=ids, namespace=self._namespace)["vectors"]).items():
-                doc = Document({"doc_id": id, "embedding": data.values} | data.metadata)  # type: ignore
+                doc_id = id.str.split()[1] if len(id.str.split()) > 1 else id
+                if data.sparse_vector:
+                    term_frequency = dict(zip(data.sparse_vector.indices, data.sparse_vector.values))
+                    data.metadata["term_frequency"] = term_frequency
```

Comment:
the `term_frequency` transform sticks this in a property so suggest `metadata['properties.term_frequency']`